agent/engine/task_manager.go

```diff
@@ -1559,3 +1569,37 @@ func (mtask *managedTask) waitForStopReported() bool {
 	}
 	return taskStopped
 }
+
+func (mtask *managedTask) UnstageVolumes(csiClient csiclient.CSIClient ) error {
+	task := mtask.Task
+	if task == nil {
+		return fmt.Errorf("managed task is nil")
+	}
+	if !task.IsEBSTaskAttachEnabled() {
+		logger.Debug("Task is not EBS-backed. Skip NodeUnstageVolume.")
+		return nil
+	}
+	for _, tv := range task.Volumes {
+		switch tv.Volume.(type) {
+		case *taskresourcevolume.EBSTaskVolumeConfig:
+			ebsCfg := tv.Volume.(*taskresourcevolume.EBSTaskVolumeConfig)
+			volumeId := ebsCfg.VolumeId
+			hostPath := ebsCfg.Source()
+			err := mtask.unstageVolumeWithTimeout(csiClient, volumeId, hostPath)
+			if err != nil {
+				logger.Error("Unable to unstage volume", logger.Fields{
+					"Task":  task.String(),
+					"Error": err,
+				})
+				return err
```

Comment:
Can there ever be multiple volumes and should we then proceed to cleanup other ones while accumulating errors?